mypy/fastparse.py

```diff
@@ -283,9 +284,15 @@ def do_func_def(self, n: Union[ast35.FunctionDef, ast35.AsyncFunctionDef],
                 # for ellipsis arg
                 if (len(func_type_ast.argtypes) == 1 and
                         isinstance(func_type_ast.argtypes[0], ast35.Ellipsis)):
+                    if n.returns:
+                        # PEP 484 disallows both type annotations and type comments
+                        self.fail(messages.DUPLICATE_TYPE_SIGNATURES, n.lineno, n.col_offset)
                     arg_types = [a.type_annotation if a.type_annotation is not None else AnyType()
                                  for a in args]
                 else:
+                    # PEP 484 disallows both type annotations and type comments
+                    if n.returns or any(a.type_annotation for a in args):
```

Comment:
It's slightly better to explicitly check that `a.type_annotation is not None` than just to check the truthiness here.